File: frontend/src/api/stats.ts
import { apiClient } from './client';
import { Article } from '../types';

export interface DailyTrend {
  date: string;
  views: number;
  visitors: number;
}

export interface TopArticleStat {
  id: string;
  title: string;
  category: string;
  views: number;
  status: string;
}

export interface CategoryStat {
  id: string;
  name: string;
  slug: string;
  article_count: number;
  percentage: number;
}

export interface DeviceStat {
  mobile_pct: number;
  desktop_pct: number;
  tablet_pct: number;
}

export interface AdminStats {
  total_articles: number;
  published_articles: number;
  draft_articles: number;
  archived_articles: number;
  featured_articles: number;
  breaking_articles: number;
  total_categories: number;
  active_categories: number;
  total_media: number;
  total_media_size_bytes: number;
  total_ads?: number;
  active_ads?: number;
  total_visitors?: number;
  today_visitors?: number;
  live_active_readers?: number;
  daily_trends?: DailyTrend[];
  top_articles?: TopArticleStat[];
  category_distribution?: CategoryStat[];
  device_breakdown?: DeviceStat;
  recent_articles: Article[];
}

export const statsApi = {
  getAdminStats: async (): Promise<AdminStats> => {
    const response = await apiClient.get<AdminStats>('/admin/stats');
    return response.data;
  },
};

File: backend/app/schemas/stats.py
from pydantic import BaseModel
from typing import List, Optional
from app.schemas.article import ArticleResponse


class DailyTrend(BaseModel):
    date: str
    views: int
    visitors: int


class TopArticleStat(BaseModel):
    id: str
    title: str
    category: str
    views: int
    status: str


class CategoryStat(BaseModel):
    id: str
    name: str
    slug: str
    article_count: int
    percentage: float


class DeviceStat(BaseModel):
    mobile_pct: int
    desktop_pct: int
    tablet_pct: int


class AdminStatsResponse(BaseModel):
    total_articles: int
    published_articles: int
    draft_articles: int
    archived_articles: int
    featured_articles: int
    breaking_articles: int
    total_categories: int
    active_categories: int
    total_media: int
    total_media_size_bytes: int
    total_ads: int
    active_ads: int
    total_visitors: int
    today_visitors: int
    live_active_readers: int
    daily_trends: List[DailyTrend]
    top_articles: List[TopArticleStat]
    category_distribution: List[CategoryStat]
    device_breakdown: DeviceStat
    recent_articles: List[ArticleResponse]
